nodes/persistence-node/src/services/CacheRunner.ts

```diff
@@ -26,48 +33,96 @@ export class CacheRunner {
     this.deps = deps;
   }
 
-  //TODO: runned missed events while updating
-  async runForPastBlocks(blockCnt: number) {
-    const latestBlock = await this.deps.ethersProvider.getBlockNumber();
+  async startIndexing(fromBlock: number) {
 
-    if(blockCnt !== 0) {
-      this.deps.logger.log("Processing past blocks...");
+    if(fromBlock > this.deps.storage.lastBlockNumber) {
+      this.deps.storage.lastBlockNumber = fromBlock;
+      await this.deps.storage.save();
+    } 
+    this.deps.logger.log(`Indexing events from block ${this.deps.storage.lastBlockNumber}...`);
 
-      await this.processPastBlocks(latestBlock - blockCnt);
-    }
-  }
+    return new Promise(async () => {
```

Comment:
Do we need to create a new promise here?

If async executor function ever threw an error, promise wouldn't get rejected, but would stay in pending state instead.